find.go

```diff
@@ -15,29 +15,57 @@ import (
 func (s *server) find(w http.ResponseWriter, r *http.Request) {
 	combined := make(chan *model.FindResponse, len(s.servers))
 	wg := sync.WaitGroup{}
+
+	// Copy the original request body in case it is a POST batch find request.
+	var rb []byte
+	var err error
+	rb, err = io.ReadAll(r.Body)
+	_ = r.Body.Close()
+	if err != nil {
+		log.Warnw("failed to read original request body", "err", err)
+		return
+	}
+
 	for _, server := range s.servers {
 		wg.Add(1)
 		go func(server *url.URL) {
 			defer wg.Done()
-			r.URL.Host = server.Host
-			r.URL.Scheme = server.Scheme
-			resp, err := s.Client.Get(r.URL.String())
+
+			// Copy the URL from original request and override host/schema to point
+			// to the server.
+			endpoint := *r.URL
+			endpoint.Host = server.Host
+			endpoint.Scheme = server.Scheme
+			log := log.With("backend", endpoint)
+
+			// If body in original request existed, make a reader for it.
+			var body io.Reader
+			if len(rb) > 0 {
+				body = bytes.NewBuffer(rb)
```

Comment:
we copy a new request body for each server?

Reply:
request takes an `io.reader`. Is there a more efficient way to pass around io.reader copies across multiple goroutines?

Reply:
I am assuming the request body is re-read by each goroutine client sending the request. No?

Reply:
you should be able to make separate readers over the same underlying byte slice https://pkg.go.dev/bytes#NewReader